OMCompiler/SimulationRuntime/c/simulation/solver/perform_simulation.c.inc

```diff
@@ -390,6 +390,19 @@ int prefixedName_performSimulation(DATA* data, threadData_t *threadData, SOLVER_
       int success = 0;
       threadData->currentErrorStage = ERROR_SIMULATION;
 
+      /* Check if loggin should be activated or deactivated */
+      if ((simInfo->useLoggingTime == 1) &&
+          (solverInfo->currentTime >= simInfo->loggingTimeRecord[0] || solverInfo->currentTime + solverInfo->currentStepSize >= simInfo->loggingTimeRecord[0]) &&
+          (solverInfo->currentTime + solverInfo->currentStepSize < simInfo->loggingTimeRecord[1]))
+      {
+        reactivateLogging();
+      }
+      if ((simInfo->useLoggingTime == 1) &&
+          (solverInfo->currentTime > simInfo->loggingTimeRecord[1]))
+      {
+        deactivateLogging();
```

Comment:
Do you call this in each step?

Reply:
Yes. Since we don't know the next step size in advance this ensures we don't miss a time interval to log stuff.